server/deploy/src/main/scala/com/prisma/deploy/schema/mutations/DeployMutation.scala

```diff
@@ -107,27 +106,8 @@ case class DeployMutation(
     }
   }
 
-  private def validateFunctionInputs(fns: Vector[FunctionInput]): Future[Vector[SchemaError]] = Future.sequence(fns.map(validateFunctionInput)).map(_.flatten)
-
-  private def validateFunctionInput(fn: FunctionInput): Future[Vector[SchemaError]] = {
-    val ProjectId(name, stage) = project.projectId
-    dependencies.graphQlClient
-      .sendQuery(
-        query = s"""{
-       |  validateSubscriptionQuery(query: ${JsString(fn.query).toString()}){
-       |    errors
-       |  }
-       |}""".stripMargin,
-        path = s"/$name/$stage/private",
-        headers = Map("Authorization" -> s"Bearer ${project.secrets.headOption.getOrElse("empty")}")
-      )
-      .map { response =>
-        response.bodyAs[Vector[String]]("data.validateSubscriptionQuery.errors").get
-      }
-      .map { errorMessages =>
-        errorMessages.map(error => SchemaError(`type` = "Subscription", field = fn.name, description = error))
-      }
-  }
+  private def validateFunctionInputs(fns: Vector[FunctionInput]): Future[Vector[SchemaError]] =
+    Future.sequence(fns.map(dependencies.functionValidator.validateFunctionInput(project, _))).map(_.flatten)
```

Comment:
@mavilein this change move the validation in process. Do you agree with this?
ref: https://github.com/graphcool/prisma/blob/master/server/api/src/main/scala/com/prisma/subscriptions/schema/SubscriptionQueryValidator.scala